File: subroutines/local-global/tax_rates.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0
# https://isaaccomputerscience.org/concepts/prog_sub_local_and_global

def calculate_basic_tax_rate():
    """Calculate tax at the basic rate"""
    amount = float(input("£"))
    tax = amount * 0.2
    print(tax)  # You could format this to two decimal places using print(f"£{tax:.2f}")  


def calculate_higher_tax_rate():
    """Calculate tax at the higher rate"""
    amount = float(input("£"))
    tax = amount * 0.4
    print(tax)    

def main():
    print("Enter an amount in pounds to find the basic tax rate:")
    calculate_basic_tax_rate()
    print("Enter an amount in pounds to find the higher tax rate:")
    calculate_higher_tax_rate()


if __name__ == '__main__':
    main() 

File: subroutines/local-global/c-sharp/tax_rates.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0
Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if neccesary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{
    class LocalGlobal
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            CalculateBasicTax();
            CalculateHigherTax();
        }
        

        // Calculate basic rate tax
        public static void CalculateBasicTax() {
            Console.Write("£ ");
            string amount = Console.ReadLine();
            double amountNumeric = double.Parse(amount);
            double tax = amountNumeric * 0.2;
            Console.WriteLine($"Basic rate tax is {tax}");
        }


        // Calculate higher rate tax
        public static void CalculateHigherTax() {
            Console.Write("£ ");
            string amount = Console.ReadLine();
            double amountNumeric = double.Parse(amount);
            double tax = amountNumeric * 0.4;
            Console.WriteLine($"Higher rate tax is {tax}");
        }
        
        
    }
}
